source/base/timer.cc

```diff
@@ -175,51 +181,48 @@ double Timer::stop ()
       rusage usage;
       getrusage (RUSAGE_SELF, &usage);
       const double dtime = usage.ru_utime.tv_sec + 1.e-6 * usage.ru_utime.tv_usec;
-      cumulative_time += dtime - start_time;
+      last_lap_cpu_time = dtime - start_time;
 
       rusage usage_children;
       getrusage (RUSAGE_CHILDREN, &usage_children);
       const double dtime_children =
         usage_children.ru_utime.tv_sec + 1.e-6 * usage_children.ru_utime.tv_usec;
-      cumulative_time += dtime_children - start_time_children;
+      last_lap_cpu_time += dtime_children - start_time_children;
 
       struct timeval wall_timer;
       gettimeofday(&wall_timer, nullptr);
       last_lap_time = wall_timer.tv_sec + 1.e-6 * wall_timer.tv_usec
                       - start_wall_time;
 #elif defined(DEAL_II_MSVC)
       last_lap_time = windows::wall_clock() - start_wall_time;
-      cumulative_time += windows::cpu_clock() - start_time;
+      last_lap_cpu_time = windows::cpu_clock() - start_time;
 #else
 #  error Unsupported platform. Porting not finished.
 #endif
 
 #ifdef DEAL_II_WITH_MPI
+      this->mpi_data = Utilities::MPI::min_max_avg (last_lap_time,
+                                                    mpi_communicator);
```

Comment:
Why do you move the global communication out of `sync_wall_time`? Isn't this going to disturb the numbers by the MPI network? I agree that in order to get the overall wall time, one needs to communicate, but isn't this supposed to be done through the `sync_wall_time` passed to the respective constructor? (I know that it will often not matter when using `MPI_COMM_SELF`.)